__tests__/integration.test.js

```diff
@@ -121,6 +122,136 @@ describe("Endpoint integration", () => {
+			return request(app)
+				.get("/api/articles?topic=mitch")
+				.expect(200)
+				.then(({ body }) => {
+					const { articles } = body;
+					expect(articles).toBeInstanceOf(Array);
+					expect(articles).toHaveLength(11);
+					articles.forEach((article) => {
+						expect.objectContaining({
+							author: expect.any(String),
+							title: expect.any(String),
+							topic: "mitch",
+							created_at: expect.any(String),
+							votes: expect.any(Number),
+							comment_count: expect.any(Number),
+						});
+					});
+				});
+		});
+		test("should accept topic as query and filter the topics by specified value", () => {
+			return request(app)
+				.get("/api/articles?topic=cats")
+				.expect(200)
+				.then(({ body }) => {
+					const { articles } = body;
+					expect(articles).toBeInstanceOf(Array);
+					expect(articles).toHaveLength(1);
+					articles.forEach((article) => {
+						expect.objectContaining({
+							author: expect.any(String),
+							title: expect.any(String),
+							topic: "cats",
+							created_at: expect.any(String),
+							votes: expect.any(Number),
+							comment_count: expect.any(Number),
+						});
+					});
+				});
+		});
+	});
+	describe("9. GET /api/articles/:article_id/comments", () => {
+		test(`should respons with an array of comments for the given article_id.
+		Each comment should have the following properties:
+		comment_id
+		votes
+		created_at
+		author
+		body`, () => {
+			return request(app)
+				.get("/api/articles/1/comments")
+				.expect(200)
+				.then(({ body }) => {
+					const { comments } = body;
+					expect(comments).toBeInstanceOf(Array);
+					expect(comments).toHaveLength(11);
+					expect(comments).toBeSortedBy("created_at", {
+						descending: true,
+					});
+					comments.forEach((comment) => {
+						expect(comment).toEqual({
+							comment_id: expect.any(Number),
+							votes: expect.any(Number),
+							created_at: expect.any(String),
+							author: expect.any(String),
+							body: expect.any(String),
+						});
+					});
+				});
+		});
+	});
+	describe("10. POST /api/articles/:article_id/comments", () => {
+		test(`should post a comment to specified article_id,
+		the body should accept and object with following properties:
+		username
+		body.`, () => {
+			const newComment = {
+				username: "butter_bridge",
+				body: "There's no shame in fear, my father told me, what matters is how we face it.",
+			};
+			return request(app)
+				.post("/api/articles/1/comments")
+				.send(newComment)
+				.expect(201)
+				.then(({ body }) => {
+					const { insertedComment } = body;
+					expect(insertedComment).toEqual({
+						article_id: 1,
+						comment_id: expect.any(Number),
+						votes: expect.any(Number),
+						created_at: expect.any(String),
+						author: "butter_bridge",
+						body: "There's no shame in fear, my father told me, what matters is how we face it.",
+					});
+				});
+		});
+	});
```

Comment:
Should see the following tests of this POST request.
  - [ ] Status 400, invalid ID, e.g. string of "not-an-id"
  - [ ] Status 404, non existent ID, e.g. 0 or 9999
  - [ ] Status 400, missing required field(s), e.g. no username or body properties
  - [ ] Status 404, username does not exist
  - [ ] Status 201, ignores unnecessary properties